### PR TITLE
Code Review Standards for External Contributors

### PR DESCRIPTION
**Pull request by JoshuaBatty**
This PR proposes a set of code review standards to be followed when external contributors submit pull requests to FuelLabs repositories. The aim of this PR is to provide a reference point for external contributors to follow.

Currently, I have the maximum line limit set at 500, and that 2 internal contributors need to approve the PR. Happy to workshop these numbers.

### CODE REVIEW

File: text/code-standards/external-contributors.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Code Review Standards for External Contributors
+
+1. __Code Line Limit__: All pull requests should not exceed 500 lines of code. This helps to ensure that changes are small and easy to review, making it less likely for bugs to be introduced and easier to understand the changes. It also keeps the pull requests manageable in size and reduces the time needed for reviewing and approving.
```
Comment (mitchmindtree) [inline text/code-standards/external-contributors.md:3]:
```suggestion
1. __Code Line Limit__: Pull requests should aim to not exceed 500 lines of additions. This helps to ensure that changes are small and easy to review, making it less likely for bugs to be introduced and easier to understand the changes. It also keeps the pull requests manageable in size and reduces the time needed for reviewing and approving.
```

As much as I dislike reviewing giant PRs, sometimes they're unavoidable for refactors, especially if there's loads of tests etc. Maybe this is better?

**Review by mitchmindtree — COMMENTED** [review]
Looks good! Perhaps we should comment on our position on `unsafe` code in this too? Perhaps we can add a "Unsafe Code" section with something along these lines:

*Safety > Performance. Correctness of code must be guaranteed before `unsafe` code will be considered for review. Requests for the inclusion of `unsafe` code should be accompanied by benchmarks that demonstrate a measurable performance improvement, and/or code comments that clearly explain why the `unsafe` code is necessary, before the code is considered for review.*

File: text/code-standards/external-contributors.md (hunk 1 above)
Comment (JoshuaBatty) [reply text/code-standards/external-contributors.md:3]: Yeah, that wording is nicer. Thanks!

**Comment by JoshuaBatty** [comment]
Yeah good point that's definitely something worth mentioning in this document. I'll add that in now. Thanks!

File: text/code-standards/external-contributors.md (hunk 1 above)
Comment (mohammadfawaz) [reply text/code-standards/external-contributors.md]: Could we also exclude tests from the line count? It's very easy to get to 500 lines with a few e2e tests (especially if JSON ABI files are also added).

**Comment by sezna** [comment]
I have no objections to what's here -- like it so far. Will give it some time to get some more eyes on it.

**Comment by luizstacio** [comment]
No objections on my side, looks good so far.

**Comment by pixelcircuits** [comment]
I'm liking this so far! No objections

**Review by otrho — COMMENTED** [review]
:+1:

**Review by sezna — APPROVED** [review]

**Review by Voxelot — APPROVED** [review]

**Review by mitchmindtree — APPROVED** [review]

**Review by mohammadfawaz — APPROVED** [review]

**Pull request merged**